core/drivers/gic.c

```diff
@@ -996,6 +1024,26 @@ static void gic_op_raise_sgi(struct itr_chip *chip, size_t it,
 	gic_it_raise_sgi(gd, it, cpu_mask, ns);
 }
 
+static void gic_it_reset(struct gic_data *gd, size_t it)
+{
+	size_t idx = it / NUM_INTS_PER_REG;
+	uint32_t mask = 1 << (it % NUM_INTS_PER_REG);
+
```

Comment:
must check that `it >= GIC_SPI_BASE`

For consistency, it think this function should also ensures the the interrupt was previously disabled. I think this means `git_it_reset()` needs a return code.

Reply:
Checked, if it < GIC_SPI_BASE, it will panic

Reply:
The API only reset the properties of interrupt, It is not necessary that interrupts have been disabled before. Functionally, the two are not coupled.

Reply:
I'm not sure it's that simple (I'm happy to be proven wrong though). What if the interrupt is active?

Reply:
OK, I will disable the irq in release flow.